src/goto-analyzer/static_verifier.cpp

```diff
@@ -19,116 +19,101 @@ Author: Martin Brain, martin.brain@cs.ox.ac.uk
 
 #include <analyses/ai.h>
 
-// clang-format off
-enum class statust { TRUE, FALSE, BOTTOM, UNKNOWN };
-// clang-format on
-
 /// Makes a status message string from a status.
-static const char *message(const statust &status)
+std::string as_string(const ai_verifier_statust &status)
 {
   switch(status)
   {
-  case statust::TRUE:
+  case ai_verifier_statust::TRUE:
     return "SUCCESS";
-  case statust::FALSE:
+  case ai_verifier_statust::FALSE_IF_REACHABLE:
     return "FAILURE (if reachable)";
-  case statust::BOTTOM:
+  case ai_verifier_statust::NOT_REACHABLE:
     return "SUCCESS (unreachable)";
-  case statust::UNKNOWN:
+  case ai_verifier_statust::UNKNOWN:
     return "UNKNOWN";
   }
   UNREACHABLE;
 }
 
-struct static_verifier_resultt
+jsont static_verifier_resultt::output_json(void) const
 {
-  statust status;
-  source_locationt source_location;
-  irep_idt function_id;
-  ai_history_baset::trace_sett unknown_histories;
-  ai_history_baset::trace_sett false_histories;
-
-  jsont output_json(void) const
-  {
-    json_arrayt unknown_json;
-    for(const auto &trace_ptr : this->unknown_histories)
-      unknown_json.push_back(trace_ptr->output_json());
-
-    json_arrayt false_json;
-    for(const auto &trace_ptr : this->false_histories)
-      false_json.push_back(trace_ptr->output_json());
-
-    return json_objectt{
-      {"status", json_stringt{message(this->status)}},
-      {"sourceLocation", json(this->source_location)},
-      {"unknownHistories", unknown_json},
-      {"falseHistories", false_json},
-    };
-  }
+  json_arrayt unknown_json;
+  for(const auto &trace_ptr : this->unknown_histories)
+    unknown_json.push_back(trace_ptr->output_json());
+
+  json_arrayt false_json;
+  for(const auto &trace_ptr : this->false_histories)
+    false_json.push_back(trace_ptr->output_json());
+
+  return json_objectt{
+    {"status", json_stringt{as_string(this->status)}},
+    {"sourceLocation", json(this->source_location)},
+    {"unknownHistories", unknown_json},
+    {"falseHistories", false_json},
+  };
+}
 
-  xmlt output_xml(void) const
-  {
-    xmlt x("result");
+xmlt static_verifier_resultt::output_xml(void) const
+{
+  xmlt x("result");
 
-    x.set_attribute("status", message(this->status));
+  x.set_attribute("status", as_string(this->status));
 
-    // DEPRECATED(SINCE(2020, 12, 2, "Remove and use the structured version"));
-    // Unstructed partial output of source location is not great...
-    x.set_attribute("file", id2string(this->source_location.get_file()));
-    x.set_attribute("line", id2string(this->source_location.get_line()));
+  // DEPRECATED(SINCE(2020, 12, 2, "Remove and use the structured version"));
+  // Unstructed partial output of source location is not great...
+  x.set_attribute("file", id2string(this->source_location.get_file()));
```

Comment:
Is this still needed?

Reply:
I know there was at least one application that was using this format so I would like to leave it at least the year we say we will leave it.